src/jit/lsra.cpp

```diff
@@ -7952,6 +7952,17 @@ void LinearScan::resolveLocalRef(BasicBlock* block, GenTreePtr treeNode, RefPosi
         if (interval->assignedReg != nullptr && interval->assignedReg->assignedInterval == interval)
         {
             interval->assignedReg->assignedInterval = nullptr;
+#ifdef _TARGET_ARM_
+            // Update overlapping floating point register for TYP_DOUBLE
+            if (interval->registerType == TYP_DOUBLE)
+            {
+                assert(genIsValidDoubleReg(interval->assignedReg->regNum));
+
+                RegRecord* nextPhysRegRecord = findAnotherHalfRegRec(interval->assignedReg);
+
+                nextPhysRegRecord->assignedInterval = nullptr;
+            }
+#endif
```

Comment:
It might be nice to extract this, perhaps into a method called `ResetAssignedReg(RegRecord, RegisterType)`. With a bit of refactoring it could also be used in `unassignPhysReg()`